### ISSUE TITLE
Prompt user to pull a base library

### ISSUE THREAD

**Issue by anovstrup** [unisonweb/unison#1336]
Instead of just hinting that the user should manually pull a base library, it would be nice if ucm would prompt the user and handle the usual cases automatically (pulling the default base or a user-specified library into `.base`). I propose this prompt (but suggestions are welcome!):

<pre>
We recommend setting up a base library before you begin. 🏗 Please enter the URL </br>of your preferred base library, press enter to accept the default, or enter <b>i</b> to </br>ignore this recommendation and setup a library later with `pull url namespace`. 

[default: https://github.com/unisonweb/base, <b>[i]</b>gnore]>
</pre>

See also #1056.

**Comment by runarorama** [comment]
Better yet, just pull it. If it goes wrong, tell the user how to do it later (and maybe retry next time). And have a `--nobase` switch to start up with out a base library.

**Comment by anovstrup** [comment]
@runarorama that works, too! I assumed that the reason it didn't already do that was to surface the choice of base library. (Also, it's maybe more courteous to let the user know before pulling data off the web? 🤷‍♂️)

**Comment by aryairani** [comment]
Yeah I thought we wanted to surface the choice of base library.  Although maybe it is annoying to start with nothing.

Maybe instead of a command line flag to not pull base, we could have a command line flag that pulls a supplied url to a supplied prefix?

`ucm -pull http://x .base http://y .runar`?

**Comment by runarorama** [comment]
I'd super prefer to not have the user jump through any hoops to set up Unison the way most people want it.

**Comment by pchiusano** [comment]
Agree with @runarorama, just pull it. It can say: “Looks like your repository is empty, so I’m pulling the base library (which you are free to delete or rename afterwards if you like).”

A modal after you launch is not really much different than documenting that UCM does this and that there’s a no-base flag.

**Comment by anovstrup** [comment]
If it's just going to always pull the default base, should it do that on `ucm init`, then, rather than waiting for you to launch ucm in interactive mode?

**Comment by ceedubs** [comment]
ucm pulls base into new codebases by default now. There are a couple of flags to tweak behavior, such as `--no-base`. There is almost certainly room for improvement (such as the namespace that it is pulled into), but let's create new tickets to address those as they come up.
